### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by Jeddaa**
N+1 problem solved. Find attached the before and after images
Screenshot 2023-07-24 at 21 23 27

After:
Screenshot 2023-07-24 at 22 52 08

- Used Capybara to write integration tests for each view.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class PostsController < ApplicationController
   before_action :set_current_user, only: %i[new create]
 
   def index
-    @posts = Post.where(author_id: params[:user_id])
     @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id])
```
Comment (mckent05) [inline app/controllers/posts_controller.rb]: - Good job done so far, but you have not exactly fixed the N+1 query. Please consider using the line of code below:
`@user.posts.includes(:comments)`

File: spec/features/posts/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      visit "/users/#{@user1.id}/posts/#{@post1.id}"
+      expect(page).to have_content('My first rejection')
+    end
+
+    it 'should redirect to user1\'s show page when user is clicked.' do
+      click_link 'Post #1'
+      expect(page).to have_content('post 2 title')
+    end
+  end
+end
```
Comment (mckent05) [inline spec/features/posts/post_index_spec.rb]: - There is a failing test in this file, please fix all tests and ensure they all pass 👍 👍 
33y3yw

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Jeddaa, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding integration tests to your project 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Jeddaa) [reply app/controllers/posts_controller.rb:6]: hello @mckent05 using this increases the number of queries my db has to run. it doesn't decrease as expected

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @Jeddaa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**